### PR TITLE
[bugfix](hive)Partition fields in wrong order

### PR DESCRIPTION
**Pull request by wuwenchi**
## Proposed changes

Wrong order of partition fields will lead to wrong order of actual partitions.
When converting doris chema to hive spartiton, it needs to be passed according to the `partition by` field order.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/PartitionTableInfo.java [hunk 1]
```diff
@@ -191,6 +194,27 @@ public void validatePartitionInfo(
                         "Duplicated partition column " + duplicatesKeys.get(0));
             }
 
+            if (engineName.equals(CreateTableInfo.ENGINE_HIVE)) {
+                // 1. Not all columns can be partition fields
+                // 2. The partition field must be at the end of the schema
+                // 3. The order of partition fields in the schema
+                //    must be consistent with the order defined in `PARTITIONED BY LIST()`
+                if (partitionColumns.size() == columns.size()) {
+                    throw new AnalysisException("Not all columns can be partition fields.");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/PartitionTableInfo.java:203]:
```suggestion
                    throw new AnalysisException("Cannot set all columns as partitioning columns.");
```

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**